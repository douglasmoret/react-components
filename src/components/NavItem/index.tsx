import React from 'react'
import classNames from 'classnames'
import Icon, { IconSlugsType } from '../Icon'
import Typography from '../Typography'
import style from './style.module.scss'

interface NavItemProps {
  className?: string
  icon: IconSlugsType
  label: string
  url: string
  onClick: () => void
  transparent?: boolean
  target?: '_blank' | '_self' | '_parent' | '_top'
  isActive: boolean
  element?: React.ElementType

}

const NavItem: React.FC<NavItemProps> = ({
  className,
  icon,
  label,
  url,
  onClick,
  children,
  transparent,
  target = '_self',
  isActive,
  element: Element,
  ...rest
}) => {
  const AnchorElement = Element || (url ? 'a' : 'span')
  const itemClassNames = classNames(style.navItem, className, {
    [style.transparent]: transparent,
  })

  if (children) {
    return <li className={itemClassNames}>{children}</li>
  }

  return (
    <li className={itemClassNames}>
      <AnchorElement
        onClick={onClick}
        target={target}
        className={style.navItemAnchor}
        href={url}
        {...rest}
      >
        <Icon className={style.navItemIcon} icon={icon} />
        <Typography
          element="span"
          variant="buttonMD"
          className={classNames(style.navItemLabel, {
            [style.navItemActive]: isActive,
          })}
        >
          {label}
        </Typography>
      </AnchorElement>
    </li>
  )
}

export default NavItem
