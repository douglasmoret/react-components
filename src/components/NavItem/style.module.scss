.navItem {
  margin: 0;
  display: flex;
}

.navItemAnchor {
  cursor: pointer;
  text-decoration: none;
  color: var(--color-primary-400);
  display: flex;
  align-items: center;
  height: var(--size-nav-item-height);
  text-transform: uppercase;
}

.navItemAnchor:hover {
  color: var(--color-primary-500);
}

.navItemIcon {
  margin-right: var(--spacing-sm);
}

.navItemIcon svg {
  fill: var(--color-primary-600);
}

.navItem .navItemLabel {
  align-self: center;
  color: var(--color-neutral-dark-300);
  text-decoration: none;
  font-family: var(--font-family);
  font-size: var(--font-size-xs);
  font-weight: var(--font-weight-bold);
  margin: unset;
}

.navItemActive {
  position: relative;
}

.navItemActive::after {
  content: '';
  width: 100%;
  height: 2px;
  display: block;
  background-color: var(--color-secondary-200);
  position: absolute;
  margin-top: calc(var(--spacing-sm) / 2);
}

@media only screen and (min-width: 1024px) {
  .navItemIcon {
    display: none;
  }

  .navItemAnchor {
    padding: var(--spacing-sm) var(--spacing-md);
  }

  .navItem .navItemLabel {
    padding-left: unset;
    transition: color 200ms linear;
    font-size: var(--font-size-xxs);
  }

  .navItemLabel:hover {
    color: var(--color-primary-500);
  }

  .navItemActive::after {
    background-color: var(--color-primary-300);
  }

}
