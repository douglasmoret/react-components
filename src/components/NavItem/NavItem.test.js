/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import NavItem from './index'

test('NavItem: simple render test', () => {
  const component = renderer.create(
    <NavItem url="#" label="baixe o app Ahazou" icon="mono_arrow_down" active target="_blank" />,
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
