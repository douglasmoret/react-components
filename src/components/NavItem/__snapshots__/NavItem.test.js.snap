// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem: simple render test 1`] = `
<li
  className="navItem"
>
  <a
    active={true}
    className="navItemAnchor"
    href="#"
    target="_blank"
  >
    <span
      className="typography navItemLabel buttonMD"
    >
      baixe o app Ahazou
    </span>
  </a>
</li>
`;
